### ISSUE TITLE
Add support for drawing infinite repeating textures with offset

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#991]
## Enhancement request:
**tl;dr** add a way to draw infinite repeating textures to screen, with an offset. 

### What should be added/changed?

A camera-sensitive way to draw a repeating texture in the background, using the player's position to offset the texture drawing.  This should probably be implemented somewhere using lower level OpenGL calls rather than `Sprite` objects.

Based on [discord discussion](https://discord.com/channels/458662222697070613/458662458198982676/895129822416732231) with @einarf , this could take the form of a `TextureGrid` object. It could have `repeat_x` and `repeat_y` parameters that specify how many times a passed pattern or individual texture should repeat in a given direction. The default could be infinite for ease of use.

This ticket does not necessarily include integration with loading from Tiled or other map sources, and that may be a separate ticket.

### What would it help with?

**tl;dr** provides an easy way to have easy, instant, and infinite non-interactive background for top-down games

For beginners who want to make top down games with restricted view, one of the easiest ways to get started is to have the camera always center on the player. This frees beginners from the difficulties common to writing good camera behavior that is aware of environments.

However, a solid-color background will not clearly show that the player's character actually moving in the world. There have to be external visual cues aside from player sprite animation to show that player characters are actually moving. Adding support for a non-sprite infinite background pattern allows the following:
1. Allow beginners to get started quickly
2. Avoid overhead for creating sprite objects for background layers
3. Improve developer experiences by enabling quicker iteration
4. Provide a default "empty" grid background for games that generate terrain

**Comment by einarf** [comment]
Closing this as a duplicate for now
